app/routes.py

```diff
@@ -26,8 +26,8 @@ def menu_search():
     session['establs'] = establishments
     return render_template('menu-search.html', title='Menu Search', categories=categories.keys(), cuisines=cuisines.keys(), establishments=establishments.keys())
 
-@app.route('/menu-browse', methods=['POST'])
-def menu_browse():
+@app.route('/search-menu-browse', methods=['POST'])
```

Comment:
don't change the route of the endpoint here. Instead, have it also support at GET request in which it gets the currently logged in user and queries for their saved menus that way.